### PR TITLE
Add All and Every function

### PR DESCRIPTION
**Pull request by pearl2201**
## Checklist for PR

[x] Link to corresponding Issue. [5](https://github.com/thecasualcoder/godash/issues/5)
[x] Tests covering all the code branches.
[x] Validations for all the argument's type. This includes the functions passed as arguments.
[x] A Comment on how the function works and what are the different validations in the function.
[x] An ExampleTest of the function.
[x] README.md should be updated about the function.

### CODE REVIEW

File: all.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// 3. Predicate function's argument should be of the same type as the elements of the input slice
+//
+// Validation errors are returned to the caller
+func All(in, predicateFn interface{}) (bool, error) {
+	var output bool = true;
+	input := reflect.ValueOf(in)
+	predicate := reflect.ValueOf(predicateFn)
+
+	if predicate.Kind() != reflect.Func {
+		return output, fmt.Errorf("predicateFn has to be a function")
```
Comment (dineshba) [inline all.go]: During error, i think it is better to return false

File: README.md [hunk 2]
```diff
@@ -169,11 +170,38 @@ func main() {
...
+
+### All or Every 
+
+All or Every checks if predicate returns truthy for all element of collection. Iteration is stopped once predicate returns falsely. 
+For more [docs](https://godoc.org/github.com/thecasualcoder/godash#All). 
+
+```go 
+func main() { 
+	input := []int{1, 2, 3, 4, 5} 
+	var output []int
```
Comment (dineshba) [inline README.md]: output is of type `bool`

File: all.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		for i := 0; i < input.Len(); i++ {
+			arg := input.Index(i)
+			returnValue := predicate.Call([]reflect.Value{arg})[0]
+			if !returnValue.Bool() {
+				output = false
+				break
+			}
+		}
+
+		return output, nil
```
Comment (dineshba) [inline all.go]: we may not need `output` variable at all. We can directly `return false, nil` at line 52 and `return true, nil` at line 57. What do you think @pearl2201 ?

**Review by dineshba — CHANGES_REQUESTED** [review]

**Comment by pearl2201** [comment]
I just commit the change.

File: README.md (hunk 2 above)
Comment (dineshba) [reply README.md]: can you make it plain bool `not []bool`

File: README.md [hunk 4]
```diff
@@ -169,11 +170,38 @@ func main() {
...
+
+### All or Every 
+
+All or Every checks if predicate returns truthy for all element of collection. Iteration is stopped once predicate returns falsely. 
+For more [docs](https://godoc.org/github.com/thecasualcoder/godash#All). 
+
+```go 
+func main() { 
+	input := []int{1, 2, 3, 4, 5} 
+	var output []bool
```
Comment (dineshba) [inline README.md]: should be just bool `not []bool`

**Review by dineshba — CHANGES_REQUESTED** [review]

**Review by dineshba — APPROVED** [review]

**Pull request merged**